velox/functions/lib/Re2Functions.cpp

```diff
@@ -730,8 +730,6 @@ void re2SplitAll(
     const re2::StringPiece remaining = input.substr(pos);
     arrayWriter.add_item().setNoCopy(
         StringView(remaining.data(), remaining.size()));
-  } else if (pos == input.size()) {
```

Comment:
Looks relevant to test failure. Looks this `if` branch is necessary. I thought wrongly.